test/validation-error-handling.test.js

```diff
@@ -97,6 +97,38 @@ test('should be able to use setErrorHandler specify custom validation error', t
   })
 })
 
+test('validation error has 400 statusCode set', t => {
+  t.plan(3)
+
+  const fastify = Fastify()
+
+  fastify.setErrorHandler((error, request, reply) => {
+    const errorResponse = {
+      message: error.message,
+      statusCode: error.statusCode || 500
+    }
+
+    reply.code(errorResponse.statusCode).send(errorResponse)
+  })
+
+  fastify.post('/', { schema }, echoBody)
```

Comment:
Could we have a test with a throwing endpoint to ensure both cases?

```js
fastify.get('/error',{},()=> throw "foo")
  fastify.inject({
    method: 'GET',
    url: '/error'
  }, (err, res) => {
    t.error(err)
    t.same(res.json(), {
      statusCode: 500,
      message: "foo"
    })
    t.equal(res.statusCode, 400)
  })
```

Reply:
The above snippet does not have any validation, so it will result in a 500.

Reply:
This is already included multiple times.

```js
fastify.get('/error',{},()=> throw "foo")

  fastify.inject({
    method: 'GET',
    url: '/error'
  }, (err, res) => {
    t.error(err)
    t.same(res.json(), {
      statusCode: 500,
      message: "foo"
    })
    t.equal(res.statusCode, 500) // note the 500 instead of 400
  })
```

Note that [this test](https://github.com/fastify/fastify/blob/1f0dc84605d14dfd4c92d8aad7b3ba2c8c9af136/test/reply-error.test.js#L449-L477) covers the above case and more.

Reply:
Correct